test/CMakeLists.txt

```diff
@@ -31,7 +33,7 @@ foreach(test ${tests})
   add_executable(${test} EXCLUDE_FROM_ALL ${test}.cc test.cc)
   set_target_properties(${test} PROPERTIES
     COMPILE_DEFINITIONS VALHALLA_SOURCE_DIR="${VALHALLA_SOURCE_DIR}/")
-  target_link_libraries(${test} valhalla)
+  target_link_libraries(${test} valhalla gtest)
```

Comment:
too bad gtest isnt header only..

Reply:
why is it bad in this case? 
We include gtest as a submodule and build it here (it seems to have only cmake and c++11 compiler as requirements)
`add_subdirectory(${VALHALLA_SOURCE_DIR}/third_party/googletest ${CMAKE_BINARY_DIR}/googletest)`

Reply:
oh no sorry its not bad, i was just surprized it wasnt header only. tbh ive never read up on gtest so maybe it has some more complicated bits in there that make header only too much of a pain.

Reply:
[Catch](https://github.com/catchorg/Catch2) is a nice header-only C++ test framework, which is trivial to use. However, `gtest`'s mocking capabilities make it a more compelling option, IMO.